Hi @yrodiere , the checkstyle didn't pass—there's an unused import:

```
[ERROR] /home/travis/build/yrodiere/hibernate-search/jsr352/core/src/main/java/org/hibernate/search/jsr352/logging/impl/Log.java:10:15: Unused import - org.jboss.logging.Logger.Level.ERROR. [UnusedImports]
```

--- a/jsr352/core/src/main/java/org/hibernate/search/jsr352/logging/impl/Log.java
+++ b/jsr352/core/src/main/java/org/hibernate/search/jsr352/logging/impl/Log.java
@@ -7,7 +7,6 @@
 package org.hibernate.search.jsr352.logging.impl;
 
 import static org.jboss.logging.Logger.Level.DEBUG;
-import static org.jboss.logging.Logger.Level.ERROR;
 import static org.jboss.logging.Logger.Level.INFO;
 import static org.jboss.logging.Logger.Level.TRACE;
 
@@ -99,30 +98,6 @@ public interface Log extends org.hibernate.search.util.logging.impl.Log {
 			value = "%1$s"
 	)
 	void analyzeIndexProgress(String progress);
-
-	@LogMessage(level = ERROR)
-	@Message(id = JSR_352_MESSAGES_START_ID + 11,
-			value = "Unable to close session."
-	)
-	void unableToCloseSession(@Cause Exception e);
-
-	@LogMessage(level = ERROR)
-	@Message(id = JSR_352_MESSAGES_START_ID + 12,
-			value = "Unable to close stateless session."
-	)
-	void unableToCloseStatelessSession(@Cause Exception e);
-
-	@LogMessage(level = ERROR)
-	@Message(id = JSR_352_MESSAGES_START_ID + 13,
-			value = "Unable to close scrollable results."
-	)
-	void unableToCloseScrollableResults(@Cause Exception e);
-
-	@LogMessage(level = ERROR)
-	@Message(id = JSR_352_MESSAGES_START_ID + 14,
-			value = "Unable to close entity manager."
-	)
-	void unableToCloseEntityManager(@Cause Exception e);
 
 	@LogMessage(level = INFO)
 	@Message(id = JSR_352_MESSAGES_START_ID + 15,

--- a/jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/impl/steps/afterchunk/AfterChunkBatchlet.java
+++ b/jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/impl/steps/afterchunk/AfterChunkBatchlet.java
@@ -62,11 +62,6 @@ public class AfterChunkBatchlet extends AbstractBatchlet {
 
 	@Override
 	public void stop() throws Exception {
-		try {
-			session.close();
-		}
-		catch (Exception e) {
-			log.unableToCloseSession( e );
-		}
+		session.close();
 	}
 }

--- a/jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/impl/steps/beforechunk/BeforeChunkBatchlet.java
+++ b/jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/impl/steps/beforechunk/BeforeChunkBatchlet.java
@@ -75,11 +75,6 @@ public class BeforeChunkBatchlet extends AbstractBatchlet {
 
 	@Override
 	public void stop() throws Exception {
-		try {
-			session.close();
-		}
-		catch (Exception e) {
-			log.unableToCloseSession( e );
-		}
+		session.close();
 	}
 }

--- a/jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/impl/steps/lucene/EntityReader.java
+++ b/jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/impl/steps/lucene/EntityReader.java
@@ -36,6 +36,7 @@ import org.hibernate.search.jsr352.massindexing.impl.util.PartitionBound;
 import org.hibernate.search.jsr352.massindexing.impl.util.PersistenceUtil;
 import org.hibernate.search.jsr352.massindexing.impl.util.SerializationUtil;
 import org.hibernate.search.util.StringHelper;
+import org.hibernate.search.util.impl.Closer;
 import org.hibernate.search.util.logging.impl.LoggerFactory;
 
 import static org.hibernate.search.jsr352.massindexing.MassIndexingJobParameters.CACHE_MODE;
@@ -185,22 +186,14 @@ public class EntityReader extends AbstractItemReader {
 	 */
 	@Override
 	public void close() throws Exception {
-		log.closingReader( serializedPartitionId, entityName );
-		try {
-			scroll.close();
-		}
-		catch (Exception e) {
-			log.unableToCloseScrollableResults( e );
-		}
-		try {
-			session.close();
-		}
-		catch (Exception e) {
-			log.unableToCloseSession( e );
-		}
 		// reset the chunk work count to avoid over-count in item collector
 		PartitionContextData partitionData = (PartitionContextData) stepContext.getTransientUserData();
 		stepContext.setPersistentUserData( partitionData );
+		try ( Closer<Exception> closer = new Closer<>() ) {
+			log.closingReader( serializedPartitionId, entityName );
+			closer.push( scroll::close );
+			closer.push( session::close );
+		}
 	}
 
 	/**

--- a/jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/impl/steps/lucene/PartitionMapper.java
+++ b/jsr352/core/src/main/java/org/hibernate/search/jsr352/massindexing/impl/steps/lucene/PartitionMapper.java
@@ -120,14 +120,10 @@ public class PartitionMapper implements javax.batch.api.partition.PartitionMappe
 	public PartitionPlan mapPartitions() throws Exception {
 
 		JobContextData jobData = (JobContextData) jobContext.getTransientUserData();
-		Session session = null;
-		StatelessSession ss = null;
+		emf = jobData.getEntityManagerFactory();
 
-		try {
-			emf = jobData.getEntityManagerFactory();
-			session = PersistenceUtil.openSession( emf, tenantId );
-			ss = PersistenceUtil.openStatelessSession( emf, tenantId );
-
+		try (Session session = PersistenceUtil.openSession( emf, tenantId );
+				StatelessSession ss = PersistenceUtil.openStatelessSession( emf, tenantId ) ) {
 			Integer maxResults = null;
 			if ( StringHelper.isNotEmpty( serializedMaxResultsPerEntity ) ) {
 				maxResults = SerializationUtil.parseIntegerParameter( MAX_RESULTS_PER_ENTITY, serializedMaxResultsPerEntity );
@@ -186,20 +182,6 @@ public class PartitionMapper implements javax.batch.api.partition.PartitionMappe
 			log.partitionsPlan( partitionPlan.getPartitions(), partitionPlan.getThreads() );
 			return partitionPlan;
 		}
-		finally {
-			try {
-				ss.close();
-			}
-			catch (Exception e) {
-				log.unableToCloseStatelessSession( e );
-			}
-			try {
-				session.close();
-			}
-			catch (Exception e) {
-				log.unableToCloseSession( e );
-			}
-		}
 	}
 
 	private List<PartitionBound> buildPartitionUnitsFrom(StatelessSession ss,
@@ -218,13 +200,12 @@ public class PartitionMapper implements javax.batch.api.partition.PartitionMappe
 		if ( maxResults != null ) {
 			criteria.setMaxResults( maxResults );
 		}
-
-		ScrollableResults scroll = criteria.setProjection( Projections.id() )
+		criteria.setProjection( Projections.id() )
 				.setFetchSize( fetchSize )
-				.setReadOnly( true )
-				.scroll( ScrollMode.FORWARD_ONLY );
+				.setReadOnly( true );
 
-		try {
+
+		try ( ScrollableResults scroll = criteria.scroll( ScrollMode.FORWARD_ONLY ) ) {
 			/*
 			 * The scroll results are originally positioned *before* the first element,
 			 * so we need to scroll rowsPerPartition + 1 positions to advanced to the
@@ -245,16 +226,6 @@ public class PartitionMapper implements javax.batch.api.partition.PartitionMappe
 			upperID = null;
 			partitionUnits.add( new PartitionBound( javaClass, lowerID, upperID, indexScope ) );
 			return partitionUnits;
-		}
-		finally {
-			try {
-				if ( scroll != null ) {
-					scroll.close();
-				}
-			}
-			catch (Exception e) {
-				log.unableToCloseScrollableResults( e );
-			}
 		}
 	}
 

